From 8a65f0e9fc450baad76d6a00453dc74d20a7d771 Mon Sep 17 00:00:00 2001
From: Bernhard Voelker <mail@bernhard-voelker.de>
Date: Tue, 29 Mar 2022 01:06:50 +0200
Subject: [PATCH 15/17] find: port to Ubuntu 20.04 cross-compiling for OpenWRT
 21.02

In the above build environment, HAVE_ENDPWENT seems to be defined to 0
instead of being #undef'ed.  Hence the build fails:

  parser.c: In function 'parse_user':
  parser.c:75:20: error: expected expression before ')' token
     75 | # define endpwent ()
        |                    ^
  parser.c:2463:7: note: in expansion of macro 'endpwent'
   2463 |       endpwent ();
        |       ^~~~~~~~

* find/parser.c (HAVE_ENDGRENT,HAVE_ENDPWENT): Change from #ifndef to
"#if !", and define the replacement code to "(void) 0".

Fixes https://github.com/openwrt/packages/issues/17912

(cherry picked from commit fa7e628e19c75a3650b9010a37f3388758245b68)
---
 find/parser.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/find/parser.c b/find/parser.c
index 5d59a885..a1d9e698 100644
--- a/find/parser.c
+++ b/find/parser.c
@@ -69,12 +69,12 @@
 #include "findutils-version.h"
 #include "system.h"
 
-
-#ifndef HAVE_ENDGRENT
-# define endgrent ()
+#if ! HAVE_ENDGRENT
+# define endgrent() ((void) 0)
 #endif
-#ifndef HAVE_ENDPWENT
-# define endpwent ()
+
+#if ! HAVE_ENDPWENT
+# define endpwent() ((void) 0)
 #endif
 
 static bool parse_accesscheck   (const struct parser_table*, char *argv[], int *arg_ptr);
-- 
2.41.0

